pkg/graveler/graveler.go

```diff
@@ -1335,14 +1335,17 @@ func (g *KVGraveler) Get(ctx context.Context, repositoryID RepositoryID, ref Ref
 	if reference.StagingToken != "" {
 		// try to get from staging, if not found proceed to committed
 		value, err := g.getFromStagingArea(ctx, reference.Branch, key)
-		if err != nil && !errors.Is(err, ErrNotFound) {
-			return nil, err
-		}
-		if value == nil {
-			// tombstone - the entry was deleted on the branch => doesn't exist
-			return nil, ErrNotFound
+		if err != nil {
+			if !errors.Is(err, ErrNotFound) {
+				return nil, err
+			} // proceed to committed
+		} else { // In staging
+			if value == nil {
+				// tombstone - the entry was deleted on the branch => doesn't exist
+				return nil, ErrNotFound
+			}
+			return value, nil
```

Comment:
Did we have a bug here too? Let's open an issue, fix and add many tests to this very critical flow. The fact that it was discovered in a system test means we are lacking coverage.

Reply:
yes, @N-o-Z is handling it on a separate PR

Reply:
We're not lacking in coverage - the unit tests are just old, there's a scenario that tests it, the mocks were just not configured correctly